examples/ui-demo/src/components/shared/MintCard.tsx

```diff
@@ -0,0 +1,241 @@
+"use client";
+import Image from "next/image";
+import { CheckIcon } from "../icons/check";
+import { GasIcon } from "../icons/gas";
+import { DrawIcon } from "../icons/draw";
+import { ReceiptIcon } from "../icons/receipt";
+import React, { useCallback, useState } from "react";
+import { hexToRGBA } from "../../utils/hexToRGBA";
+import { LoadingIcon } from "../icons/loading";
+import { ExternalLinkIcon } from "../icons/external-link";
+import {
+  useSendUserOperation,
+  useSmartAccountClient,
+} from "@account-kit/react";
+import { AccountKitNftMinterABI, nftContractAddress } from "@/utils/config";
+import { encodeFunctionData } from "viem";
+
+type NFTLoadingState = "loading" | "success";
+
+const initialState = {
+  signing: "signing",
+  gas: "gas",
+  batch: "batch",
+} satisfies mintStatus;
+
+type mintStatus = {
+  signing: NFTLoadingState | "signing";
+  gas: NFTLoadingState | "gas";
+  batch: NFTLoadingState | "batch";
+};
+
+export const MintCard = () => {
+  const [status, setStatus] = useState<mintStatus>(initialState);
+  // To be wired into the toast pr
+  const [hasError, setHasError] = useState(false);
+  const [hasCollected, setHasCollected] = useState(false);
+  const handleSuccess = () => {
+    setStatus((prev) => ({ ...prev, batch: "success" }));
+    setHasCollected(true);
+  };
+  const { client } = useSmartAccountClient({ type: "LightAccount" });
+  const handleError = () => {
+    setStatus(initialState);
+    setHasError(true);
+  };
+  const { sendUserOperationResult, sendUserOperation } = useSendUserOperation({
+    client,
+    waitForTxn: true,
+    onError: handleError,
+    onSuccess: handleSuccess,
+  });
+
+  const getPrimaryColorRGBA = useCallback(() => {
+    if (typeof window === "undefined") return hexToRGBA("#363FF9", 0.1);
+    const color = getComputedStyle(document.documentElement)
+      .getPropertyValue("--akui-fg-accent-brand")
+      .trim();
+    const rgba = hexToRGBA(color, 0.1);
+    return rgba;
+  }, []);
+
+  const handleCollectNFT = useCallback(async () => {
+    if (!client) {
+      console.error("no client");
+      return;
+    }
+    setStatus({
+      signing: "loading",
+      gas: "loading",
+      batch: "loading",
+    });
+    await new Promise<void>((resolve) => {
+      setTimeout(() => {
+        setStatus((prev) => ({ ...prev, signing: "success" }));
+        resolve();
+      }, 2000);
+    });
+    await new Promise<void>((resolve) => {
+      setTimeout(() => {
+        setStatus((prev) => ({ ...prev, gas: "success" }));
+        resolve();
+      }, 2000);
+    });
```

Comment:
this should probably go in the `onMutate` function of the `useSendUserOperation` hook. 

Right now this would block the call to `sendUserOperation` for 4 seconds is that intended?

Reply:
The intention was to simulate the stages resolving one by one as per design's instruction.  We can adjust this to resolve at the same time.

Reply:
ah yea that makes sense, we should keep the sequential signing + gas order, but just fire those things `onMutate` and have the `onSuccess` mark them all as successful. This should make things feel a bit faster.

We also might want to tweak signing timing to be like 150-200ms because that's how fast the signatures api typically is and for gas sponsorhip similarly 500ms-1s potentially

Reply:
Implemented